coderd/database/dbauthz/dbauthz.go

```diff
@@ -458,6 +471,13 @@ func As(ctx context.Context, actor rbac.Subject) context.Context {
 		// should be removed from the context.
 		return context.WithValue(ctx, authContextKey{}, nil)
 	}
+	return InjectActorToContext(ctx, actor)
+}
+
+func InjectActorToContext(ctx context.Context, actor rbac.Subject) context.Context {
+	if rlogger := loggermw.RequestLoggerFromContext(ctx); rlogger != nil {
+		rlogger.WithAuthContext(actor.ID, actor.FriendlyName, actor.Email, actor.Type)
```

Comment:
Honestly you could just pass in the whole `rbac.Subject` object here, I don't think there's an import loop or anything preventing that, and it avoids you having to make your own type for it